pkg/filter/wasm/wasm.go

```diff
@@ -0,0 +1,317 @@
+// +build wasmfilter
+
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package wasm
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/megaease/easegress/pkg/cluster"
+	"github.com/megaease/easegress/pkg/context"
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/object/httppipeline"
+)
+
+const (
+	// Kind is the kind of WasmFilter.
+	Kind        = "WasmFilter"
```

Comment:
There is a filter naming `WasmFilter` that sounds weird/redundant to me, can we rename it to `WasmRuntime` or something else.

Reply:
wasmtime and wasmer are wasm runtimes, so `WasmRuntime` is not a suitable name for the filter, in fact, I tried to find another one, but none seems better.

Reply:
`WasmVM `？

Reply:
there's `WasmVM` already in our code...

Reply:
How about `Wasm`

Reply:
`Wasm` often refers to many other things like the wasm code and may bring additional confusion.
I'll rename it to `WasmHost` as it is a host environment of WebAssembly, any idea?

Reply:
I'm fine with it